src/lib/token.ts

```diff
@@ -0,0 +1,89 @@
+import { Config, TokenSearchRules, TokenOptions } from '../types'
+import { encode64 } from './utils'
+import crypto from 'crypto'
+
+/**
+ * Create the header of the token.
+ *
+ * @param {String} apiKey API key used to sign the token.
+ * @param {String} encodedHeader Header of the token in base64.
+ * @param {String} encodedPayload Payload of the token in base64.
+ * @returns {String} The signature of the token in base64.
+ */
+function sign(apiKey: string, encodedHeader: string, encodedPayload: string) {
+  return crypto
+    .createHmac('sha256', apiKey)
+    .update(`${encodedHeader}.${encodedPayload}`)
+    .digest('base64')
+    .replace(/\+/g, '-')
```

Comment:
I don't think I replace the first occurrence only. While `replace` is not `replaceAll` it does replaceAll if used with a regex.
[As per replace documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace). 

> The replace() method returns a new string with some or all matches of a pattern replaced by a replacement. The pattern can be a string or a [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp), and the replacement can be a string or a function to be called for each match. If pattern is a string, only the first occurrence will be replaced.

I avoid using `replaceAll` because it is not compatible in previous `node` version, for example in `node 14` which is still supported. See [compatibilities of features with node](https://node.green/)

Reply:
Oh, I didn't know that! Thanks for the explanation!